assets/js/hooks/carousel.js

```diff
@@ -0,0 +1,212 @@
+
+import scrollIntoView from "scroll-into-view-if-needed";
+import smoothScrollIntoView from "smooth-scroll-into-view-if-needed";
+
+export default {
+    mounted() {
+        this.reel = this.el.querySelector(".moon-reel");
+        this.items = this.reel.querySelectorAll("li");
+        this.indicators = this.el.querySelectorAll(".moon-indicator");
+        this.itemCount = this.items.length;
+        this.leftArrow = this.el.querySelector(".moon-left-arrow");
+        this.rightArrow = this.el.querySelector(".moon-right-arrow");
+        this.itemsArray = this.getArray("li", this.reel);
+        this.indicatorsArray = this.getArray(".moon-indicator", this.el);
+        this.initScrolls();
+        this.setActiveItem();
+        this.initIntersectionObserver();
+        this.el.addEventListener("triggerLeftArrowClick", () => {
+            this.handleLeftArrowClick();
+        });
+        this.el.addEventListener("triggerRightArrowClick", () => {
+            this.handleRightArrowClick();
+        }); 
+        this.delay = parseInt(this.el.dataset.autoslide_delay, 10);
+        if (!isNaN(this.delay)) {
+            this.startAutoSlide(this.delay);
+        };
+    },
+    
+    updated() {
+        this.scrollToIndex();
+    },
+
+    setActiveItem() {
+        const value = parseInt(this.el.dataset.value, 10);
+        const lastIndex = this.itemsArray.length - 1;
+        const scrollToIndex = this.itemsArray.findIndex((item, index) => index === value);
+
+        if (scrollToIndex < lastIndex && scrollToIndex !== -1) {
+             this.scrollIntoViewSmoothly(this.itemsArray[scrollToIndex], {
+                block: "nearest",
+                behavior: "smooth",
+                inline: "nearest",
+                boundary: this.reel,
+            });
+        }
+    },
+
+    initScrolls() {
+        if (document && "scrollBehavior" in document.documentElement.style) {
```

Comment:
NB: I need to double check how it actually works. I think we will discuss it on our call @tkullisaar